File: task-03/prime.java
import java.util.Scanner;

class prime 
{
    static boolean isPrime(int n)
    {
        if (n <= 1)
            return false;
  
        for (int i = 2; i < n; i++)
            if (n % i == 0)
                return false;
  
        return true;
    }

    public static void main(String[] args)
    {   
        Scanner input = new Scanner(System.in);

        System.out.print("Enter a number: ");
        int num = input.nextInt();
        for (int i = 2; i <= num; i++) {
            if (isPrime(i))
                System.out.print(i + " ");
        }
    }
}

File: task-03/prime.exs
defmodule Prime do
  def is_prime(num) when num <= 1, do: false
  def is_prime(num) when num == 2, do: true
  def is_prime(num) do
    is_prime(num, 2)
  end

  defp is_prime(num, divisor) when divisor >= num, do: true
  defp is_prime(num, divisor) when rem(num, divisor) == 0, do: false
  defp is_prime(num, divisor) do
    is_prime(num, divisor + 1)
  end

  def main do
    IO.puts("Enter any number: ")
    n = IO.gets("") |> String.trim() |> String.to_integer()

    for i <- 2..n, is_prime(i) do
      IO.write("#{i} ")
    end
    IO.puts(" ")
  end
end

Prime.main()
